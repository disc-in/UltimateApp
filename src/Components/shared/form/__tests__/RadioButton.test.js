import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { TouchableOpacity } from 'react-native';

import RadioButton from '../RadioButton';

describe('RadioButton', () => {
  const values = ['red', 'blue'];
  const labels = ['I prefer red', 'I prefer blue'];

  it('renders correctly', () => {
    const { toJSON } = render(
      <Formik initialValues={{ title: 'My title' }}>
        <RadioButton fieldName="title" label="Title" labels={labels} values={values} />
      </Formik>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when there is an error', async () => {
    const { toJSON, getByTestId, getByText } = render(
      <Formik
        initialValues={{ title: 'blue' }}
        validationSchema={Yup.object({ title: Yup.string().oneOf(['blue'], 'Invalid title') })}
      >
        {({ handleSubmit }) => (
          <>
            <RadioButton fieldName="title" label="Title" labels={labels} values={values} />
            <TouchableOpacity onPress={handleSubmit} testID="submit" />
          </>
        )}
      </Formik>,
    );

    fireEvent.press(getByText('I prefer red'));

    await waitFor(() => {
      fireEvent.press(getByTestId('submit'));
    });

    expect(getByText('Invalid title')).not.toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });
});
