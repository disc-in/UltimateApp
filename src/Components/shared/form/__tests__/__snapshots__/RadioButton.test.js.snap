// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      Object {
        "marginBottom": 5,
      }
    }
  >
    Title
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#e5e5e5",
          "borderRadius": 5,
          "borderWidth": 0.5,
          "flexBasis": "30%",
          "flexGrow": 1,
          "marginRight": 5,
          "opacity": 1,
          "paddingVertical": 10,
        }
      }
      testID="input-title-red"
    >
      <Text
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        I prefer red
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#e5e5e5",
          "borderRadius": 5,
          "borderWidth": 0.5,
          "flexBasis": "30%",
          "flexGrow": 1,
          "marginRight": 5,
          "opacity": 1,
          "paddingVertical": 10,
        }
      }
      testID="input-title-blue"
    >
      <Text
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        I prefer blue
      </Text>
    </View>
  </View>
</View>
`;

exports[`RadioButton renders correctly when there is an error 1`] = `
Array [
  <View
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "marginBottom": 5,
        }
      }
    >
      Title
    </Text>
    <Text
      style={
        Object {
          "color": "red",
          "fontStyle": "italic",
        }
      }
    >
      Invalid title
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#2a9e91",
            "borderRadius": 5,
            "borderWidth": 0.5,
            "flexBasis": "30%",
            "flexGrow": 1,
            "marginRight": 5,
            "opacity": 1,
            "paddingVertical": 10,
          }
        }
        testID="input-title-red"
      >
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          I prefer red
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#e5e5e5",
            "borderRadius": 5,
            "borderWidth": 0.5,
            "flexBasis": "30%",
            "flexGrow": 1,
            "marginRight": 5,
            "opacity": 1,
            "paddingVertical": 10,
          }
        }
        testID="input-title-blue"
      >
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          I prefer blue
        </Text>
      </View>
    </View>
  </View>,
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="submit"
  />,
]
`;
